src/components/Recorder/index.js

```diff
@@ -1,56 +1,22 @@
-import { useState } from 'react'
+import { useMemo } from 'react'
 import PropTypes from 'prop-types'
-import useMediaRecorder from '../../hooks/useMediaRecorder'
-//import Visualizer from '../Visualizer'
 import Recording from '../Recording'
 import './style.css'
+import useMediaRecorder from "../../hooks/useMediaRecorder";
 
 const Recorder = ({stream}) => {
-    const defaultRecordClass = 'record-play'
-    const [recordingStateText, setRecordingStateText] = useState('Record')
-    const [recordings, setRecordings] = useState([])
-    const [recordButtonClassesText, setRecordButtonClassesText] = useState(defaultRecordClass)
-    let chunks = []
-
-    const onStop = () => {
-
-        const blob = new Blob(chunks, { 'type' : 'audio/ogg; codecs=opus' })
-        chunks = []
-        const audioURL = window.URL.createObjectURL(blob)
-        setRecordings(recordings => {
-            return [...recordings, ...[{
-                stream: audioURL, 
-                name: new Date().toISOString().split('.')[0].split('T').join(' '),
-                id: `id${recordings.length}`
-            }]]
-        })
-        setRecordButtonClassesText(defaultRecordClass)
-        setRecordingStateText('Record')
-    }
-
-
-    const onStart = () => {
-
-    
-    }
-
-
-    const ondataavailable = (e) => {
-        chunks.push(e.data)
-    }
+    const { recorder, recordings, setRecordings, isRecording } = useMediaRecorder(stream);
 
+    const defaultRecordClass = 'record-play'
+    const recordButtonClassesText = useMemo(() => isRecording ? `${defaultRecordClass} recording-audio` : defaultRecordClass, [isRecording])
```

Comment:
Let's see if I understand why `useMemo` rather than `useState`…

If `isRecording` hasn't changed, `useMemo` won't rerun the function and will return the cached value directly. However, determining the value of `isRecording` does not seem to be an expensive operation. Are you sure `useMemo` is better here than `useState`? What am I missing?

Reply:
You are right, this is not an expensive operation, but using a combination of `useState` + `useEffect` would make the code more complex, and it would probably also cause the component to rerender more times since you would be mutating the state in two different places, you component and your hook.
But yes, it is mostly a decision based on code simplicity and readability.